cmd/metalctl/app/move.go

```diff
@@ -0,0 +1,347 @@
+	crdsToMove := make([]*apiextensionsv1.CustomResourceDefinition, 0, len(sourceCrds))
+	for _, sourceCrd := range sourceCrds {
+		targetCrd := sourceCrd.DeepCopy()
+		err := targetClient.Get(ctx, client.ObjectKeyFromObject(sourceCrd), targetCrd)
+		if apierrors.IsNotFound(err) {
+			crdsToMove = append(crdsToMove, sourceCrd)
+			continue
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to check CRD existence in the target cluster: %w", err)
+		}
+
+		if reflect.DeepEqual(clearFields(sourceCrd), clearFields(targetCrd)) {
+			slog.Debug("source and target CRDs are the same", slog.String("CRD", crdKind(sourceCrd)))
+			continue
+		}
+		return nil, fmt.Errorf("a CRD %s exists in the target cluster and is different then in the source cluster", sourceCrd.GetName())
+	}
+	return crdsToMove, nil
+}
+
+func getCrsToBeMoved(ctx context.Context, targetClient client.Client, sourceCrs []*unstructured.Unstructured) ([]*unstructured.Unstructured, error) {
+	crsToMove := make([]*unstructured.Unstructured, 0, len(sourceCrs))
+	for _, sourceCr := range sourceCrs {
+		targetCr := sourceCr.DeepCopy()
+		err := targetClient.Get(ctx, client.ObjectKeyFromObject(sourceCr), targetCr)
+		if apierrors.IsNotFound(err) || meta.IsNoMatchError(err) {
+			crsToMove = append(crsToMove, sourceCr)
+			continue
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to check CR existence in the target cluster: %w", err)
+		}
+
+		if reflect.DeepEqual(clearFields(sourceCr), clearFields(targetCr)) {
+			slog.Debug("source and target CRs are the same", slog.String("CR", crName(sourceCr)))
+			continue
+		}
+		return nil, fmt.Errorf("a CR %s/%s already exists in the target cluster and is different then in the source cluster", sourceCr.GetNamespace(), sourceCr.GetName())
+	}
+	return crsToMove, nil
+}
+
+type Node struct {
+	Cr       *unstructured.Unstructured
+	Children []*Node
+}
+
+func crsOwnerReferenceTrees(crs []*unstructured.Unstructured) []*Node {
+	nodeMap := make(map[types.UID]*Node)
+
+	for _, cr := range crs {
+		nodeMap[cr.GetUID()] = &Node{Cr: cr}
+	}
+	roots := []*Node{}
+	for _, cr := range crs {
+		if len(cr.GetOwnerReferences()) == 0 || nodeMap[cr.GetOwnerReferences()[0].UID] == nil {
+			roots = append(roots, nodeMap[cr.GetUID()])
+		} else {
+			owner := nodeMap[cr.GetOwnerReferences()[0].UID]
+			owner.Children = append(owner.Children, nodeMap[cr.GetUID()])
+		}
+	}
+	return roots
+}
+
+func cleanup[T client.Object](ctx context.Context, cl client.Client, objs []T) error {
+	cleanupErrs := make([]error, 0)
+	for _, obj := range objs {
+		if err := cl.Delete(ctx, obj); err != nil {
+			cleanupErrs = append(cleanupErrs, err)
+		}
+	}
+	return errors.Join(cleanupErrs...)
+}
+
+func moveCrds(ctx context.Context, cl client.Client, sourceCrds []*apiextensionsv1.CustomResourceDefinition) (movedCrds []*apiextensionsv1.CustomResourceDefinition, err error) {
```

Comment:
Heretic question @Nuckal777: Do we really need this? I would suggest we run a `make install` on the target cluster to install the CRDs into the target environment. That is also the way `clsuterctl` from the CAPI project is doing it. https://cluster-api.sigs.k8s.io/clusterctl/commands/move.html#pivot

Reply:
It's what you [ordered in the ticket](https://github.com/ironcore-dev/metal-operator/issues/146), but I am fine with expecting the CRDs to present before the move command is issued.

Reply:
@Nuckal777 dropping the CRD handling will make the codepath a lot easier (see comments about the usage of `unstructured`). Let me adjust the issue and I would say lets drop the CRD handling for now.

Reply:
Ok, I updated the issue. We can also drop the `--cr-only` flag as this should be the default (and only) copy operation.